### PR TITLE
[release] [statefun] Announcement blog post and Downloads page update for StateFun 3.0.0

### PR DESCRIPTION
**Pull request by tzulitai**
The release artifacts are already out. The actual announcement is currently pending on this being reviewed + merged.

### CODE REVIEW

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+For more details, check the complete [release changelog](https://issues.apache.org/jira/secure/ReleaseNote.jspa?version=12348822&styleName=&projectId=12315522) 
+and the [updated documentation](https://ci.apache.org/projects/flink/flink-statefun-docs-release-3.0/).
+We encourage you to download the release and share your feedback with the community through the [Flink mailing lists](https://flink.apache.org/community.html#mailing-lists)
+or [JIRA](https://issues.apache.org/jira/browse/)!
+
+{% toc %}
+
+## Background
+
+Starting with the very first StateFun release before the project was donated to the Apache Software Foundation, our focus was: **making scalable stateful applications easy to build and run**.
```
Comment (sjwiesman) [inline _posts/2021-04-14-release-statefun-3.0.0.md:31]:
```suggestion
Starting with the first StateFun release, before the project was donated to the Apache Software Foundation, our focus was: **making scalable stateful applications easy to build and run**. 
```

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+We encourage you to download the release and share your feedback with the community through the [Flink mailing lists](https://flink.apache.org/community.html#mailing-lists)
+or [JIRA](https://issues.apache.org/jira/browse/)!
+
+{% toc %}
+
+## Background
+
+Starting with the very first StateFun release before the project was donated to the Apache Software Foundation, our focus was: **making scalable stateful applications easy to build and run**. 
+  
+The first StateFun version introduced an SDK that allowed writing stateful functions that build up a StateFun application, which is packaged and deployed as a very specific Flink job that is submitted to a Flink cluster. Having functions executing within the same JVM as Flink has some advantages such as performance and immutability of the deployment. However, it had a few limitations:
```
Comment (sjwiesman) [inline _posts/2021-04-14-release-statefun-3.0.0.md:33]:
```suggestion
The first StateFun version introduced an SDK that allowed writing stateful functions that build up a StateFun application packaged and deployed as a particular Flink job submitted to a Flink cluster. Having functions executing within the same JVM as Flink has some advantages, such as the deployment's performance and immutability. However, it had a few limitations:
```

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+1. ❌ ⠀Functions can be written only in a JVM based language.
+2. ❌ ⠀A blocking call/CPU heavy task in one function can affect other functions and operations that need to complete in a timely manner, such as checkpointing.
+3. ❌ ⠀Deploying a new version of the function required a stateful upgrade of the backing Flink job.
+
+With StateFun 2.0.0, the debut official release after the project was donated to Apache Flink, the community introduced the concept of “remote functions”, together with an additional SDK for the Python language.
+A remote function is a function that executes in a separate process and is invoked via HTTP by Apache Flink.
+Remote functions introduces a new and exciting capability: **state and compute disaggregation** - allowing users to scale the functions independently of the Flink cluster, which essentially plays the role of handling messaging and state in a consistent and fault-tolerant manner.
```
Comment (sjwiesman) [inline _posts/2021-04-14-release-statefun-3.0.0.md:41]:
```suggestion
With StateFun 2.0.0, the debut official release after the project was donated to Apache Flink, the community introduced the concept of “remote functions,” together with an additional SDK for the Python language.
A remote function is a function that executes in a separate process and is invoked via HTTP by Apache Flink.
Remote functions introduce a new and exciting capability: **state and compute disaggregation** - allowing users to scale the functions independently of the Flink cluster, which essentially plays the role of handling messaging and state in a consistent and fault-tolerant manner.
```

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+While remote functions did address the limitations (1) and (2) mentioned above, we still had some room to improve:
+  
+1. ✅ ⠀Functions can be written in any language (initially Python)
+2. ✅ ⠀Blocking calls / CPU heavy computations happens in a separate process(es) 
+3. ❌ ⠀Registering a new function, or changing the state definitions of an existing function required a stateful upgrade of the StateFun processes.
+4. ❌ ⠀The SDK had a few friction points around state and messaging ergonomics - it had a heavy dependency on Google’s Protocol Buffers for it’s multi-language object representation.
+ 
+
+With the new **StateFun 3.0** release, the community has enhanced the remote functions protocol (the protocol that describes how StateFun communicates with the remote processes) to address the issues above.
+Building on the new protocol we rewrote the Python SDK, and introduced a brand new remote Java SDK.
```
Comment (sjwiesman) [inline _posts/2021-04-14-release-statefun-3.0.0.md:52]:
```suggestion
Building on the new protocol, we rewrote the Python SDK and introduced a brand new remote Java SDK.
```

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+    
+For a detailed SDK tutorial, we would like to encourage you to visit:
+
+- [Java SDK showcase](https://github.com/apache/flink-statefun-playground/tree/release-3.0/java/showcase)
+- [Python SDK showcase](https://github.com/apache/flink-statefun-playground/tree/release-3.0/python/showcase)
+
+## Dynamic Registration of State and Functions
+
+Starting with this release, it is now possible to dynamically register new functions, without going through a stateful upgrade cycle of the StateFun cluster (which entails the standard process of performing a stateful restart of a Flink job).
```
Comment (sjwiesman) [inline _posts/2021-04-14-release-statefun-3.0.0.md:123]:
```suggestion
Starting with this release, it is now possible to dynamically register new functions without going through a stateful upgrade cycle of the StateFun cluster (which entails the standard process of performing a stateful restart of a Flink job).
```

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+authors:
+- igalshilman:
+  name: "Igal Shilman"
+  twitter: "IgalShilman"
+- tzulitai:
+  name: "Tzu-Li (Gordon) Tai"
+  twitter: "tzulitai"
+---
+
+The Apache Flink community is happy to announce the release of Stateful Functions (StateFun) 3.0.0!  This release focuses on bringing remote functions to the front and center of StateFun. It is now easier, more efficient, and more ergonomic to write applications that live in a separate process to the StateFun cluster.
```
Comment (sjwiesman) [inline _posts/2021-04-14-release-statefun-3.0.0.md:16]: This needs a sentence or two to introduce statefun to new users coming across this release announcement

**Review by sjwiesman — CHANGES_REQUESTED** [review]

File: _posts/2021-04-14-release-statefun-3.0.0.md (hunk 6 above)
Comment (sjwiesman) [reply _posts/2021-04-14-release-statefun-3.0.0.md:16]: @knaufk maybe you have some ideas? I need to think about it

Comment (sjwiesman) [reply _posts/2021-04-14-release-statefun-3.0.0.md:16]: Something like: "Stateful Functions provides a cross-platform kit for building stateful serverless applications. This release now focuses on ..."? 

Idk

Comment (StephanEwen) [reply _posts/2021-04-14-release-statefun-3.0.0.md:16]: I think that is, we need an introduction sentence...
Drafting one now, filling in in a bit.

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+title:  "Stateful Functions 3.0.0 Release Announcement"
```
Comment (knaufk) [inline _posts/2021-04-14-release-statefun-3.0.0.md:3]:
```suggestion
title:  "Stateful Functions 3.0.0: Remote Functions Front and Center"
```
Adding the "slogan" to the title might make it more catchy.

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+  name: "Igal Shilman"
+  twitter: "IgalShilman"
+- tzulitai:
+  name: "Tzu-Li (Gordon) Tai"
+  twitter: "tzulitai"
+---
+
+The Apache Flink community is happy to announce the release of Stateful Functions (StateFun) 3.0.0!  This release focuses on bringing remote functions to the front and center of StateFun. It is now easier, more efficient, and more ergonomic to write applications that live in a separate process to the StateFun cluster.
+
+The binary distribution and source artifacts are now available on the updated [Downloads](https://flink.apache.org/downloads.html)
```
Comment (knaufk) [inline _posts/2021-04-14-release-statefun-3.0.0.md:18]: Can we move these more "technical" announcement further down in the blog post (link to artifacts and release notes)?

File: _posts/2021-04-14-release-statefun-3.0.0.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+        return context.done();
+    }
+}
+```
+
+Although there are some language specific differences, the terms and concepts are the same:
+
+* an address scoped storage acting as a key-value store for a particular address.
+* A unified cross-language way to send, receive, and store values across languages.
```
Comment (knaufk) [inline _posts/2021-04-14-release-statefun-3.0.0.md:112]:
```suggestion
* a unified cross-language way to send, receive, and store values across languages.
```

**Review by knaufk — COMMENTED** [review]
Thank you for drafting the announcement. Generally looks good. If you have the time you can try to put  even more emphasize on what the new features allow users to do: 
* what use cases or architectures do we enable, which previously have not been possible?
* why does it matter that you don't need to restart the Flink Cluster anymore to add functions or change state descriptions? 
* why does it matter to have multiple SDKs with the same concepts? Why does the polyglot nature matter?

File: _posts/2021-04-14-release-statefun-3.0.0.md (hunk 8 above)
Comment (StephanEwen) [reply _posts/2021-04-14-release-statefun-3.0.0.md:18]: +1

**Comment by StephanEwen** [comment]
What do you think about this as the introductory paragraph?

----

The Apache Flink community is happy to announce the release of Stateful Functions (StateFun) 3.0.0! 
Stateful Functions is a cross-platform stack for building Stateful Serverless applications, making it radically simpler to develop scalable, consistent, and elastic distributed applications (on Kubernetes and Clouds platforms).

This new release brings the *remote functions* to the front and center of StateFun, making the disaggregated setup that separates the application logic from the StateFun cluster the default. It is now easier, more efficient, and more ergonomic to write applications that live in their own processes or containers. With the new Java SDK this is now also possible for all JVM languages, in addition to Python.

----

File: _posts/2021-04-14-release-statefun-3.0.0.md (hunk 6 above)
Comment (tzulitai) [reply _posts/2021-04-14-release-statefun-3.0.0.md:16]: Applied @StephanEwen's proposed intro paragraph posted below.

File: _posts/2021-04-14-release-statefun-3.0.0.md (hunk 8 above)
Comment (tzulitai) [reply _posts/2021-04-14-release-statefun-3.0.0.md:18]: Moved this paragraph to a `Release Resources` section at the very end.

**Comment by StephanEwen** [comment]
If you want, feel free to drop the "on Kubernetes and Cloud platforms" part. I wasn't sure about that. It might sound limiting, but it might also help narrow the scope in a good way.

**Pull request closed without merging**